### DISCUSSION TITLE
post/url-is-strange/

### DISCUSSION THREAD

**Discussion by giscus** [kakaopay/techlog-comments#99] [Announcements]
# post/url-is-strange/

카카오페이 알림피드 서비스에서 발생한 장애 원인과 해결 과정을 공유합니다.

https://tech.kakaopay.com/post/url-is-strange/

**Comment by jws1837** [comment]
깊이 있는 원인분석과 해결 도움 많이 됐습니다.

**Reply by sinb57** [reply]
도움을 드릴 수 있어서 다행이네요🙇‍♂️

**Comment by yhkee0404** [comment]
너무 좋은 글이었습니다! [Pull Request](https://github.com/spring-projects/spring-framework/pull/33608) 작성해 봤는데 ~~괜찮으시면 도와 주실 수 있나요?~~ 좀더 큰 이슈였던 spring-projects/spring-framework#33542 의 언급과 함께 거절되었네요 😆

**Reply by sinb57** [reply]
아 Spring 프로젝트에 pr 올려주셨는데 반려됐군요.. 대신 제안 남겨주셔서 감사합니다!! 아쉽지만 라이브러리에 대한 fix 제안은 쉽지 않더라구요. 사용하는 분들이 많다보니 사이드이펙트를 예측하기 어려워서 쉽게 반영하기 어려울거예요ㅠ 관심 가져주셔서 감사합니다!!

**Reply by yhkee0404** [reply]
@sinb57 Side effect로 본문의 정규식이 사라지나 봅니다: spring-projects/spring-framework@6c62965 :scream: 기존에 `fromUriString`에서 사용하던 `UrlParser`는 정규식을 사용했지만 `6.2.0-M2`부터 RFC 3986이 아니라 WhatWG URL Living standard를 따르도록 바꾸었나 봐요: spring-projects/spring-framework@f21e05a 그 이름을 다시 `6.2.0-RC2`부터  `WhatWgUrlParser`로 바꾸고; spring-projects/spring-framework@10f4522, RFC 3986을 새로운 기본값으로 지원하기 위해 `RfcUriParser`를 만들었네요: spring-projects/spring-framework@52805da 둘다 정규식 대신 긴 구현을 택한 것이 의외입니다. 그리고 Pull Request 없이 직접 commit하시는 maintainer님이 대단하신데 이렇게도 Change history 관리가 잘 되는지 의아하네요. 좋은 글 써 주신 덕분에 많은 것을 배워갑니다. 감사합니다 👍 👍 👍

**Comment by LichKing-lee** [comment]
좋은 글 잘 봤습니다 문제를 하나씩 파악해가며 찾아가는게 인상적이에요!

**Comment by d0lim** [comment]
해결하는 과정부터, 실제로 소나큐브 룰 적용까지 너무 좋네요! 잘 보고 갑니다~

**Comment by ACACIANh** [comment]
재밌게 잘봤습니다~
